discovery_engine/test/ffi/types/string_test.dart

```diff
@@ -34,4 +35,23 @@ void main() {
     ffi.drop_string(place);
     expect(res, equals(string));
   });
+
+  test('reading written some string yields same result', () {
+    // ignore: unnecessary_cast
+    const String? string = 'a&+KLA)&+fjw)&+f' as String?;
```

Comment:
unfortunately the type declaration `const String?` is not enough, but the type cast `as String?` is required, because dart otherwise infers that this is a non-null string indeed and tries to call `StringFfi.writeNative()` instead of `OptionStringFfi.writeNative()` in l.42.

declaring this `final` instead of `const` triggers another lint, but avoids the type cast. anybody got an idea if this can be fixed? it's just an issue in this test after all due to the const nature of the variable.

Reply:
It's only horrible with the cast or is not working even with the cast?
Are `const`/`final` changing the semantic of the type inference (a part from constness)?

Reply:
it's only working with the cast, the issue is that we need the additional `// ignore: unnecessary_cast` as this lint is bugged in this case.

`final String? string` will actually convince the compiler that this type is `String?` without an additional cast needed. with `const String? string` the compiler thinks that this type is `String` without the additional cast. but declaring it final doesn't solve the linter issue, because we then have to `// ignore: prefer_const_declarations`.

Reply:
I don't think we can do much then.